### ISSUE TITLE
Discussion: features with hard costs

### ISSUE THREAD

**Issue by jakerella** [HabitRPG/habitica#5324]
I submitted an integration idea on [Trello](https://trello.com/c/jW0pKNIm) for SMS, but I wanted to open an issue because I didn't find any good information on features that might incur a hard cost (like SMS). I would think these would need to be limited to users with a subscription... thoughts? How would one proceed with getting such an idea approved? In the case above it would require an account with an SMS gateway (like Twilio) and a credit card attached.

This Github issue is not meant to discuss the SMS feature (check [Trello](https://trello.com/c/jW0pKNIm) for that), but rather to discuss any such feature. Specifically: how does a feature with costs get approved? Is there a corporate CC to be used for such things? How does a community contributor work with the HabitRPG team to implement such a feature? (FYI, I'm willing to work on such a feature as SMS because I want it! But I have no idea how to proceed.)

**Comment by Alys** [comment]
Getting a costed feature approved and integrated is a case-by-case thing. There's no set process for it.

In this particular case, I recommend that we wait to see if other people express interest in your idea on Trello. Without sufficient interest, this is unlikely to go ahead since a cost is involved (note that this does not guarantee that it would go ahead even if there is some interest). If you wish to, you could post once today in the Tavern to ask people who are interested to comment on that card, and you could post once more in a few days to attract the attention of any people who did not see the first post. 

I would be concerned about exactly what the costs would be. We have players from many countries, with varying SMS prices, and the costs for this feature could be hard to predict and/or unexpectedly high.

**Comment by jakerella** [comment]
I completely agree with your points on SMS, all valid quite valid. As for the pricing, you are correct that for International numbers it would be different all over, but for the US it could be quite cheap ($0.0075 per text with Twilio). Anyway, I can post that info on the Trello card. But I guess the answer to this particular discussion is: case-by-case... which is expected.

So, to answer this discussion thread, a good general process might be:
1. Suggest a feature on Trello
2. If there is sufficient support to warrant further investigation of costs, etc, the core team does it (?)
3. If the feature is deemed worthy, it might get added to the roadmap, or someone could work on it independently (?)
4. Once the feature is ready to move out of development, there would be some policy integration to factor it into the subscription system
5. ...?

Again, this is ore for general documentation when there are folks that want to contribute features that may need to be for subscribers only.

**Comment by jakerella** [comment]
Oh, also... who can create the actual card on Trello? Right now it's just in the "REQUEST INTEGRATION" card comments.

**Comment by deilann** [comment]
Only members of the Trello board can create new cards. Be patient. :) We're
all busy folks and even if it got its own card, it would be quite some time
before it was worked on, most likely. We've got bigger dragons to slay, you
might say.

**Comment by Alys** [comment]
As I said, the process will be a case-by-case thing. A defined process doesn't seem necessary. Whether a staff member or an established developer or a new developer investigates or creates a new feature really depends on the feature. All feature requests can be made through Trello, as you've done, and we can take it from there each time, using whichever steps are necessary at the time.
